src/components/Collections/Settings/Settings.vue

```diff
@@ -362,34 +416,39 @@ export default {
     setCurrentSelectedTab() {
       if (this.localSettings != null) {
         let firstHeaderName = [...this.localSettings.keys()][0];
-        let firstTabName = [...this.localSettings.get(firstHeaderName).keys()][0];
-        let firstTabDetails = this.localSettings.get(firstHeaderName).get(firstTabName);
+        let firstTab = this.localSettings.get(firstHeaderName);
+        let firstTabName = [...firstTab.keys()][0];
+        let firstTabDetails = firstTab.get(firstTabName);
         this.currentSelectedTab.set(firstTabName, firstTabDetails);
+        this.currentSelectedHeaderName = firstHeaderName;
       }
     },
     /**
      * Mark a tab as selected/unselected
-     * @param {String} tabName - The name of the tab that is to be marked selected
-     * @param {Object} tabDetails - The details of the tab that is to be marked selected
+     * @param {String} tabName - the name of the tab that is to be marked selected
+     * @param {Object} tabDetails - the details of the tab that is to be marked selected
+     * @param {String} headerName - name of the header to which the selected tab belongs to
      */
-    selectTab(tabName, tabDetails) {
+    selectTab(tabName, tabDetails, headerName) {
       if (this.isMobileScreen && this.isTabSelected(tabName)) {
         // in mobile view, the tabs are toggable
         // if someone clicks on an already opened tab, close it and currentSelectedTab
         // will be set to empty
-        this.currentSelectedTab.clear();
         this.currentSelectedTab = new Map();
+        this.currentSelectedHeaderName = null;
       } else {
         // mark the clicked tab as the currentSelectedTab
         this.currentSelectedTab.clear();
```

Comment:
remove `currentSelectedTab` and use currentSelectedTabName and currentSelectedTabDetails directly.